### ISSUE TITLE
Does not work

### ISSUE THREAD

**Issue by q3fuba** [ElvUI-TBC/ElvUI_RightClick#1]
hi there
The AddOn is Loaded and everything seems right.
But Right-Click Self-Cast does not work.
I've tried to disable all other AddOns and just enabled 

Also tried the Original one with this code:
```
local E, L, V, P, G = unpack(ElvUI); 
local AB = E:GetModule('ActionBars');

local function UpdateButtonConfig(self, bar, buttonName)	
	for i, button in pairs(bar.buttons) do
		button:SetAttribute("unit2", "player")  --the only modified line
	end
end
hooksecurefunc(AB, "UpdateButtonConfig", UpdateButtonConfig)
```
and also does not work...
But i have no idea why :/

But ALT-Click works fine...
Do you have any idea?

**Comment by crumdidlyumshis** [comment]
I actually just tested the Wrath version of this addon and it does exactly the same thing. The only thing I noticed that works on both versions is if if you go to Interface Settings -> Combat -> Auto Self Cast (toggle this). I am beginning to wonder if both versions of this addon don't work. Maybe I am doing something wrong.

**Comment by crumdidlyumshis** [comment]
I take that back... I was doing something wrong on Wrath version. It works there. I will keep messing with TBC version and see.

**Comment by q3fuba** [comment]
yep i know, i've also tried to set the attribute to the Bar instead of the Buttons but it does not work at all... :/
But on "Default UI" there are some AddOns that work like [rSelfCast](https://www.wowace.com/projects/project-15028) and [RightWing](https://www.wowace.com/projects/right-wing).

This is the Code that [RightWing](https://www.wowace.com/projects/right-wing) uses...
And it seems almost the same than you did!?!
```
MainMenuBarArtFrame:SetAttribute("unit2", "player"); 
MultiBarLeft:SetAttribute("unit2", "player");
MultiBarRight:SetAttribute("unit2", "player");
MultiBarBottomLeft:SetAttribute("unit2", "player");
MultiBarBottomRight:SetAttribute("unit2", "player");
BonusActionBarFrame:SetAttribute("unit2", "player");
```

**Comment by crumdidlyumshis** [comment]
@q3fuba I was just looking at those same addons before you posted this. I kinda think its something in the actionbars.lua that is causing this. If not, could be in oUF or unitframes. It is hard to tell. I will get Bunny to take a look when he's online.

**Comment by crumdidlyumshis** [comment]
fixed! a268caff878e3d265b12e3eb64390aa3d32b3787
